src/pal/src/misc/miscpalapi.cpp

```diff
@@ -375,14 +375,17 @@ PAL_Random(
             sInitializedMRand = TRUE;
         }
 
-        for (DWORD i = 0; i < dwLength; i++)
+        for (i = 0; i < dwLength; i += sizeof(long))
         {
-            if (i % sizeof(long) == 0) {
-                num = mrand48();
-            }
+            long num = mrand48();
+            memset((char *)lpBuffer + i, &num, sizeof(num));
```

Comment:
It seems you meant to use `memcpy` here instead of `memset`. Also, this code is not correct, e.g. it would overwrite `sizeof(long)-1` bytes after the end of the buffer if `dwLength` was 1.

Reply:
Ouch. Yes. I don't know what I was thinking here. I rewrote the function now; shouldn't overwrite anything.